examples/exampleFMU/exampleFMU2.f90

```diff
@@ -0,0 +1,112 @@
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!                          Futility Development Group                          !
+!                             All rights reserved.                             !
+!                                                                              !
+! Futility is a jointly-maintained, open-source project between the University !
+! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
+! can be found in LICENSE.txt in the head directory of this repository.        !
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!> @brief Program to demo basic FMU model interaction
+!>
+!> To use, Download the example third party FMU from the fmi-cross-check repo on github:
+!>   wget https://github.com/modelica/fmi-cross-check/raw/master/fmus/2.0/cs/linux64/MapleSim/2016.2/Rectifier/Rectifier.fmu
+!>
+!> Extract the FMU with unzip:
+!>   unzip -d /path/to/fmu/rectifier_example_fmu Rectifier.fmu
+!>
+!> Run the example program:
+!>   ./Futility_exampleFMU2.exe /path/to/fmu/rectifier_example_fmu
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+PROGRAM testFMU2
+  USE Strings
+  USE IntrType
+  USE ParameterLists
+  USE FMU_Wrapper
+  IMPLICIT NONE
+
+  TYPE(FMU2_Slave) :: test_fmu2_slave
+  TYPE(ParamType) :: FMU_params
+  CHARACTER(len=256) :: unzipDirectory
+  INTEGER(SIK) :: id=3_SIK
+  REAL(SRK) :: h=1.0E-6_SRK
+  REAL(SRK) :: timeStart=0.0_SRK
+  REAL(SRK) :: timeEnd=1.0E-1_SRK
+  REAL(SRK) :: tol=1.0E-9_SRK
+  REAL(SRK) :: time, v1
+  REAL(SRK) :: minsamplestep
+  INTEGER(SIK) :: i
+  TYPE(StringType) :: varName
+
+  ! Example FMU parameter settings
+  CALL FMU_params%clear()
+  CALL FMU_params%add('FMU_Wrapper->id',id)
+
+  IF (IARGC()==1) THEN
+    CALL getarg(1, unzipDirectory)
+  ELSE
+    unzipDirectory='/home/hephaestus/proj/Futility_dev/Futility/examples/exampleFMU/reference_fmus_2016'
```

Comment:
This is an example of how to use the FMU wrapper.  I do not intend to use this as a unit test due to portability concerns of the downloaded FMU.

The path here is the unzipDirectory default value, which I will remove.  A user of this script should supply the FMU unzipDir as a command line arg   via `getarg(1, unzipDirectory)`.   This example relies on a third party pre-compiled FMU from the [web address](https://github.com/modelica/fmi-cross-check/raw/master/fmus/2.0/cs/linux64/MapleSim/2016.2/Rectifier/Rectifier.fmu) given at the top of this file. I think we could download this FMU file via CMAKE commands, but I am unsure if that is the best solution.  

The downloaded third party FMU is not portable to OSX or Win, and may not work on all Linux machines either.  Therefore; the utility of this exampleFMU.f90 script is limited as a test.

I wrote a  _portable_ FMU wrapper unit test at `unit_tests/testFMU_Wrapper/testFMU_Wrapper.f90` covers the FMU wrapper fairly well (except for rewind currently, because that has to be implemented on the FMU side).

We could move this example, and the second `exampleFMU2_MSRE_HX.f90` example, to a separate repo because it is not providing a large amount of utility here.

Reply:
> Is this path going to be portable for this test? Shouldn't the cmake declaration be passing in an argument for the location? Another option is to just put the zip in the test directory and copy it over to the build directory with CMake so it is in the same directory as the test exe.

Again, this is NOT intended to be used as a unit or integration test. I do not want to include a binary blob (FMU) froma third party source inside the Futility repo, so I will not be putting the zip file here.  Instead, I've configured the example scripts to automatically fetch the FMU from the internet, unzip it, fix a formatting problem in the FMU XML, then load it.  I removed the hard-coded paths from the examples.